src/main/java/htsjdk/samtools/cram/encoding/writer/DataSeriesWriter.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * ****************************************************************************
+ * Copyright 2013 EMBL-EBI
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ****************************************************************************
+ */
+package htsjdk.samtools.cram.encoding.writer;
+
+import htsjdk.samtools.cram.encoding.BitCodec;
+import htsjdk.samtools.cram.structure.DataSeriesType;
+import htsjdk.samtools.cram.encoding.Encoding;
+import htsjdk.samtools.cram.encoding.EncodingFactory;
+import htsjdk.samtools.cram.encoding.reader.DataSeriesReader;
+import htsjdk.samtools.cram.io.BitOutputStream;
+import htsjdk.samtools.cram.io.ExposedByteArrayOutputStream;
+import htsjdk.samtools.cram.structure.EncodingParams;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * A CRAM Data Series writer for a particular Encoding, DataSeriesType and associated parameters
+ *
+ * Note: the class does not have a writeArray method like its counterpart {@link DataSeriesReader} because
+ * array length is known when writing, therefore the same interface can be used both for single objects and arrays.
+ *
+ * @param <T> data type of the series to be written.
+ */
+public class DataSeriesWriter<T> {
+    private final BitCodec<T> codec;
+    private final BitOutputStream bitOutputStream;
```

Comment:
Normally a writer type object that encloses a closable stream would have a `close()` method and extend `Closeable`, is there a reason to not do that here?

Reply:
It's trickier than that - it may use this stream or it may use one of the referenced streams in the outputMap.  That's something that I'll be changing in a future refactor.  When it properly encloses a stream, that would be a good time to make it a `Closeable`